grpc.go

```diff
@@ -310,6 +356,51 @@ func (a *agentGRPC) runProcess(cid string, agentProcess *pb.Process) (pid int, e
 	return pid, nil
 }
 
+// This function updates the container namespaces configuration based on the
+// sandbox information. When the sandbox is created, it can be setup in a way
+// that all containers will share some specific namespaces. This is the agent
+// responsibility to create those namespaces so that they can be shared across
+// several containers.
+// If the sandbox has not been setup to share namespaces, then we assume all
+// containers will be started in a new namespace.
+// The value of a.sandbox.sharedPidNs.path will always override the namespace
+// path set by the spec.
+// TODO: This needs to be handled in a better way, so that we don't ignore
+// namespace paths from the spec.
```

Comment:
I'm afraid we **SHOULD** ignore them. Those fields are used to share namespaces on the same host for docker containers. With kata agent we better fit in the kubernetes pod model that shared/private namespaces are both at a pod level. IOW, we cannot share a namespace from a container that lives in a different sandbox. If some containers shares a namespace, they should share the pod level one.

Then we can implement as @laijs suggested in #59 that the agent always creates pod level namespaces. And if a container wants to share some namespace, it joins the pod level one.

fyi, the k8s pod spec sets namespace sharing at the pod level. See https://v1-8.docs.kubernetes.io/docs/api-reference/v1.8/#podspec-v1-core.

Reply:
@bergwolf yes you're right, I was not thinking about those drawbacks and limitaitons. Then let me remove the `TODO` from this comment, and I think the PR is ready then.